modules/ldap_query/lambda/lambda.py

```diff
@@ -0,0 +1,443 @@
+        )
+
+
+def get_file_name(file_name, extension):
+    timestamp = datetime.now().strftime("%Y_%m_%d_T%H%M%S.%f")
+    return f"{file_name}_{timestamp}.{extension}"
+
+
+def create_json_doc(**content):
+    """create a table"""
+    # This can be fleshed out to make the retrieved information
+    # more user friendly if desired/required
+    artifact = {}
+    artifact["content"] = json.dumps(content["users"])
+    artifact["file_name"] = get_file_name("user_expiration_table", "json")
+    artifact["raw_scan_results"] = True
+    return artifact
+
+
+def get_html_table_headers(**content):
+    """
+    Return the keys of the first element in the user_list dict.
+
+    We don't care if the dict is unordered b/c the expectation is
+    that each element of the dict will be structured the same
+    """
+    table_headers = []
+    user_list = content["users"]
+    days_since_pwdlastset = content["days_since_pwdlastset"]
+    try:
+        for key in user_list[days_since_pwdlastset][0].keys():
+            table_headers.append(key)
+        return table_headers
+    except IndexError:
+        # return the empty list of there are no accounts to be disabled
+        return []
+
+
+def render_template(template="html_table.html", **kwargs):
+    env = Environment(
+        loader=FileSystemLoader(
+            os.path.join(os.path.dirname(__file__), "templates"), encoding="utf8"
+        )
+    )
+    template = env.get_template(template)
+    return template.render(**kwargs)
+
+
+def create_html_table(**content):
+    # un-group the users for the html table
+    template_contents = {
+        "table_headers": get_html_table_headers(**content),
+        "user_list": content["users"],
+    }
+    artifact = {}
+    artifact["content"] = render_template(**template_contents)
+    artifact["file_name"] = get_file_name("user_expiration", "html")
+    return artifact
+
+
+def generate_artifacts(**content):
+    """Returns the list of objects to upload to s3"""
+    artifacts = {}
+    artifacts["user_expiration_table"] = create_json_doc(**content)
+    artifacts["user_expiration_html"] = create_html_table(**content)
+    return artifacts
+
+
+def put_object(dest_bucket_name, dest_object_name, src_data):
+    """
+    Add an object to an Amazon S3 bucket
+    """
+    # Put the object
+    s3 = boto3.client("s3")
+    # log.debug(f"destination object name: {dest_object_name}")
+    s3.put_object(
+        Bucket=dest_bucket_name,
+        ACL="private",
+        ContentEncoding="utf-8",
+        Key=dest_object_name,
+        Body=src_data,
+    )
+
+
+def create_presigned_url(bucket_name, object_name, expiration=3600):
+    s3 = boto3.client("s3")
+    try:
+        response = s3.generate_presigned_url(
+            "get_object",
+            Params={"Bucket": bucket_name, "Key": object_name},
+            ExpiresIn=expiration,
+        )
+    except s3.exceptions.ClientError as e:
+        log.error(e)
+        return None
```

Comment:
Is it desirable for this to avoid raising the exception?

Reply:
negatory, removed